### PR TITLE
Issue64 task get route for single listing

### LINKED ISSUE #64: [TASK] Implement a GET route for retrieving a single existing listing from DB
### Associated User Story (TBD): 
#23 or/and #26 

### Task Description (TBD):
- This ticket aims to implement an API endpoint for creating new user account.
- Route: `/api/v1/listing/:id`
- Areas that involve updates:
   - `/api` folder (**required**)
   - `/controllers` folder (**required**)
   - `/service` folder (optional - TBD - may have another task ticket to refactor codes from this)
   - `/dao` folder (optional - TBD - may have another task ticket to refactor codes from this)
   - `/dto` folder (optional - TBD - may have another task ticket to refactor codes from this)

### Dependencies:
N/A

### Acceptance Criteria (TBD):
- `/api/v1/listing/:id` returns a single JSON object, which consists of fields as follows:
   1. Id,
   2. Title, 
   3. Price, 
   4. Number of bedrooms, 
   5. Number of bathrooms, and 
   6. Description.

### PR DESCRIPTION
**Pull request by AzizulHakim96**
### Associated issues:
User story: #23 or #26
dev task: #64 

### Task Description:
This ticket aims to implement an API endpoint to retrieve one specified listing from db.

### GitHub Action
closes #64 

Success!
get1correct

Invalid Id
get1

### CODE REVIEW

File: server/controllers/listing.js [hunk 1]
```diff
@@ -19,13 +19,26 @@ class ListingController {
...
     }
   };
+
+  getOneListing = async (req, res, next) => {
+    try{
+      const listingId = req.params.id;
+      const listing = await this.listingService.getOneListing(req.params.id);
+      if(listing == null){
+        next(ApiError.notFound(`Listing with id ${listingId} is not found.`));
+        return;
```
Comment (NathanCarriere) [inline server/controllers/listing.js:35]: I believe there was a style change for these types of functions in #75 where the empty return statements were removed

File: server/controllers/listing.js [hunk 2]
```diff
@@ -19,13 +19,26 @@ class ListingController {
...
-   
-      res.status(200).json(listing);
+      const listings = await this.listingService.getAllListings(req.params);
+      res.status(200).json(listings);
     } catch (err) {
       next(ApiError.internal(`${err}`));
     }
   };
+
+  getOneListing = async (req, res, next) => {
```
Comment (NathanCarriere) [inline server/controllers/listing.js:29]: I'm thinking that just `getListing` would be more consistent with other names (such as addListing)

**Review by NathanCarriere — APPROVED** [review]

**Review by Mik3Qiao — APPROVED** [review]

File: server/controllers/listing.js (hunk 1 above)
Comment (RonnieDsouza) [reply server/controllers/listing.js:35]: Yep! @AzizulHakim96 just remove the retuen on line 35!

File: server/controllers/listing.js (hunk 2 above)
Comment (RonnieDsouza) [reply server/controllers/listing.js:29]: I agree :) getListing implies it's one Listing

**Review by RonnieDsouza — CHANGES_REQUESTED** [review]
Looks good! Just remove the extra returns mentioned and change thee name to ```getListing``` instead of ```getOneListing```. I'll approve it then. 👍

File: server/controllers/listing.js [hunk 3]
```diff
@@ -26,13 +26,13 @@ class ListingController {
...
-  getOneListing = async (req, res, next) => {
+  getListing = async (req, res, next) => {
     try{
       const listingId = req.params.id;
-      const listing = await this.listingService.getOneListing(req.params.id);
+      const listing = await this.listingService.getListing(req.params.id);
       if(listing == null){
         next(ApiError.notFound(`Listing with id ${listingId} is not found.`));
-        return; 
+        return;
```
Comment (RonnieDsouza) [inline server/controllers/listing.js:35]: We have started to remove these ```return```s in the controllers. After a ```next()```. I mentioned this in my last review too btw 😛  But just this one line change and that's it!

**Comment by RonnieDsouza** [comment]
Btw @AzizulHakim96 there's also a conflict. Let me know if you'd like me to help out! We can hop on a 5 min call and knock this PR out!

File: server/controllers/listing.js (hunk 3 above)
Comment (AzizulHakim96) [reply server/controllers/listing.js:35]: Yeah, I'm facing a problem though! If I remove this, the error message doesn't print properly

Comment (AzizulHakim96) [reply server/controllers/listing.js:35]: withoutReturn

Comment (AzizulHakim96) [reply server/controllers/listing.js:35]: Think you have to zoom a little bit to see the error.

Comment (RonnieDsouza) [reply server/controllers/listing.js:35]: Ah I see. I think the logic order may be the issue then. Just leave it in for now then.

Comment (RonnieDsouza) [reply server/controllers/listing.js:35]: There is still a merge conflict tho, so lemme know if you need help resolving that!

**Review by RonnieDsouza — APPROVED** [review]

**Pull request merged**